### PR TITLE
Add Floating Feedback Button and Customizable Page Titles

### PR DESCRIPTION
**Pull request by Baicheng-MiQ**
## What type of PR is this?

Please check the category that applies:

- [x] Feature
- [ ] Bug Fix
- [ ] Refactor
- [x] Documentation Update

## Description

This pull request introduces two main enhancements to the OpenFeedback plugin:

1. **Floating Feedback Button:**  
   - Added the ability to render the feedback button as a floating action button (FAB) anywhere on the page. This is customizable using the `floating` prop and can be positioned with inline styles.
   - Updated `Root.tsx` to demonstrate the floating button implementation with a feedback icon.

2. **Customizable Page Titles and Subtitles:**  
   - Modified the `OpenFeedbackPage` component to allow passing custom titles and subtitles via props. This enables more flexibility when integrating the feedback page into different sections of the Backstage app.
   - Included unit tests to verify that the custom title and subtitle are correctly rendered.

## Related Issues

(No specific issues linked)

## Testing

- [x] I have included or updated unit tests for my changes.
- [x] I would like some help writing tests.
- [ ] These changes do not require tests.

### CODE REVIEW

**Comment by Parsifal-M** [comment]
Hey! 👋 

Some cool stuff in here! 🚀 

Thanks a lot!

Will take a look 👀

**Comment by Baicheng-MiQ** [comment]
added changes from https://github.com/Parsifal-M/backstage-open-feedback/pull/48#issuecomment-2265308474

**Comment by Parsifal-M** [comment]
> added changes from [#48 (comment)](https://github.com/Parsifal-M/backstage-open-feedback/pull/48#issuecomment-2265308474)

Thanks a lot!

I have to run for the day now 😢 

Will try give it a proper review later today/weekend and get back to you!

Thanks again!  🚀

File: plugins/open-feedback/src/components/OpenFeedbackModal/OpenFeedbackModal.tsx [hunk 1]
```diff
@@ -20,22 +20,26 @@ import {
   FormControlLabel,
   Checkbox,
   DialogActions,
+  Fab,
 } from '@material-ui/core';
+import { cleanUrl } from '../../utils/cleanUrl';
 
-export type SidebarOpenfeedbackProps = {
+export type ButtonOpenfeedbackProps = {
```
Comment (Parsifal-M) [inline plugins/open-feedback/src/components/OpenFeedbackModal/OpenFeedbackModal.tsx:27]: Looking at this now, there's no need to export it, and I am not sure we should make a `types.ts` file for the take of one type...

Lets just do `type ButtonOpenfeedbackProps = {...` 

👍

File: plugins/open-feedback/src/components/OpenFeedbackModal/OpenFeedbackModal.tsx [hunk 2]
```diff
@@ -20,22 +20,26 @@ import {
...
   icon?: IconComponent;
+  floating?: boolean;
+  style?: React.CSSProperties;
```
Comment (Parsifal-M) [inline plugins/open-feedback/src/components/OpenFeedbackModal/OpenFeedbackModal.tsx:30]: I think we need to do two things here to keep it more customisable for everyone, lets add a `color?` and `title?` to these props like:

```ts
type ButtonOpenfeedbackProps = {
  icon?: IconComponent;
  floating?: boolean;
  style?: React.CSSProperties;
  title?: string;
  color?: 'primary' | 'secondary';
};
```

Then we give people the option to have the modal fit their Backstage app color theme and can give it a fancy title if they like! Then we can just pass it as:

```ts
        <Fab
          color={props.color ?? 'primary'}
          variant="extended"
          onClick={() => setOpen(true)}
          style={props.style}
        >
          <Icon style={{ marginRight: 4 }} />
          {props.title}
        </Fab>
 ```

And of then of course update the readme to say you can do this:

```ts
    <OpenFeedbackModal
      title="Super Feedback!"
      floating
      color='secondary'
      icon={FeedbackIcon}
      style={{ position: 'fixed', bottom: 20, right: 20, color: 'white' }}
    />
```

Hope you don't think I am being too picky! I just think its a really nice to have!

WDYT?

File: plugins/open-feedback/src/components/OpenFeedbackPage/OpenFeedbackPage.tsx [hunk 3]
```diff
@@ -3,18 +3,23 @@ import { Typography, Grid } from '@material-ui/core';
...
 
-export const OpenFeedbackPage = () => (
+export interface OpenFeedbackPageProps {
+  title?: string;
+  subtitle?: string;
+}
+
+export const OpenFeedbackPage = ({
+  title = 'Welcome to OpenFeedback!',
+  subtitle = 'Collected Feedback for your Backstage App!',
```
Comment (Parsifal-M) [inline plugins/open-feedback/src/components/OpenFeedbackPage/OpenFeedbackPage.tsx:13]: Would it be okay to add a small snippet in the README on how you'd update the title and subtitle to something else if you want in the `App.tsx` like:

```ts
    <Route path="/open-feedback" element={<OpenFeedbackPage title='My Super Feedback Title!' subtitle='A Super Subtitle!'/>} />
 ```

WDYT? 😅

**Review by Parsifal-M — COMMENTED** [review]
Very nice! Super happy with this ⭐ 

Thanks again for taking the time and interest in contributing! 🚀

File: packages/app/src/components/Root/Root.tsx [hunk 4]
```diff
@@ -94,5 +95,10 @@ export const Root = ({ children }: PropsWithChildren<{}>) => (
       </SidebarGroup>
     </Sidebar>
     {children}
+    <OpenFeedbackModal
+      floating
+      icon={FeedbackIcon}
+      style={{ position: 'fixed', bottom: 20, right: 20, color: 'white' }}
```
[the remark this answers is not in the document]
Comment (Baicheng-MiQ) [reply packages/app/src/components/Root/Root.tsx]: Sure! https://github.com/Parsifal-M/backstage-open-feedback/pull/49/commits/e54afc8de468a35902e6000a9a2499ebb1a0fc0c

File: plugins/open-feedback/src/components/OpenFeedbackModal/OpenFeedbackModal.tsx [hunk 5]
```diff
@@ -20,22 +20,26 @@ import {
...
   const [comment, setComment] = useState('');
   const [anonymous, setAnonymous] = useState(false);
   const [open, setOpen] = useState(false);
-  const [url, setUrl] = useState(window.location.href);
+  const [url, setUrl] = useState(cleanUrl(window.location.href));
   const feedbackApi = useApi(openFeedbackBackendRef);
   const identity = useApi(identityApiRef);
   const Icon = props.icon ? props.icon : ThumbUpAltIcon;
-
+  const Floating = props.floating ?? false;
```
[the remark this answers is not in the document]
Comment (Baicheng-MiQ) [reply plugins/open-feedback/src/components/OpenFeedbackModal/OpenFeedbackModal.tsx]: Thank you for spotting that! https://github.com/Parsifal-M/backstage-open-feedback/pull/49/commits/f5d9cd04a11003e5ca295d79abfdd86e1becd956

File: README.md [hunk 6]
```diff
@@ -100,6 +102,18 @@ import { OpenFeedbackModal } from '@parsifal-m/backstage-plugin-open-feedback';
...
 
+Floating button
+
+```typescript
+import { OpenFeedbackModal } from '@parsifal-m/backstage-plugin-open-feedback';
+
+// Anywhere
+<OpenFeedbackModal
+  floating
+  style={{ position: 'fixed', bottom: 20, right: 20 }}
```
[the remark this answers is not in the document]
Comment (Baicheng-MiQ) [reply README.md]: Yes! https://github.com/Parsifal-M/backstage-open-feedback/pull/49/commits/0b7429289e98b1f180b79addb5b500175aeaaea2

File: plugins/open-feedback/src/components/OpenFeedbackModal/OpenFeedbackModal.tsx (hunk 1 above)
Comment (Baicheng-MiQ) [reply plugins/open-feedback/src/components/OpenFeedbackModal/OpenFeedbackModal.tsx]: thank you for pointing that out! Yes you're right, just changed that: https://github.com/Parsifal-M/backstage-open-feedback/pull/49/commits/59a7d447b9623b238d33f7004bb901a372aef98b

File: plugins/open-feedback/src/components/OpenFeedbackModal/OpenFeedbackModal.tsx (hunk 2 above)
Comment (Baicheng-MiQ) [reply plugins/open-feedback/src/components/OpenFeedbackModal/OpenFeedbackModal.tsx:30]: Not picky at all! I really appreciate the detailed review and feedback. It’s my pleasure to have my work reviewed by a senior engineer! https://github.com/Parsifal-M/backstage-open-feedback/pull/49/commits/642880f5f94f2238fae66f2edf40876e511d5bef

**Comment by Parsifal-M** [comment]
LGTM! 

Well Done! Awesome stuff 🚀 

Will try set some time aside this week to release this 😄 

Thanks again!

**Review by Parsifal-M — APPROVED** [review]

**Comment by Baicheng-MiQ** [comment]
Brilliant!!
Happy to help!
Thank you so much for the review! 🥇

**Pull request merged**